modules/tinymce/src/plugins/accordion/main/ts/core/Keyboard.ts

```diff
@@ -0,0 +1,64 @@
+import Editor from 'tinymce/core/api/Editor';
+import VK from 'tinymce/core/api/util/VK';
+
+import * as Utils from './Utils';
+
+const setupEnterKeyInSummary = (editor: Editor): void => {
+  editor.on('keydown', (event): void => {
+    if (event.shiftKey) {
+      return;
+    }
+    if (event.keyCode !== VK.ENTER) {
+      return;
+    }
+    if (!Utils.isInSummary(editor)) {
+      return;
+    }
+    event.preventDefault();
+    editor.execCommand('ToggleAccordion');
+  });
+};
+
+const setupEnterKeyInAccordionBody = (editor: Editor): void => {
```

Comment:
Is it really necessary? I do not understand why it should be called on `insertParagraph` as well. Also is there really any other application for this function except the keyboard keypress?

Also it would make the code a bit messier, since we need to prevent the default event and thus this function must accept a callback or return a boolean or something like that.

Reply:
I guess we could discuss this but currently there is similar code in the core and that code can be simulated using the commands. See this fiddle for an example: https://fiddle.tiny.cloud/A5iaab/1 the blockquote will be exited if you hit enter twice or if you issue the command twice the same code is running in the core.

So I guess we have four options:
1. Don't do the command logic. Then `InsertParagraph` will not be the same as a `enter` keypress
2. Do the command override, would involve some prevent defaults and hack in the plugin for those commands effected.
3. Move the enter code to the core so that any enter in the last empty block in a `details` element exits the `details` element similar to `blockquote`.
4. At least prepare for 2 but do 1 so that the code is separated and isolated from keyboard logic so that it can in theory be executed as a command.

It would have been nice if we had some form of enter registry of functions for this but it's so seldom we need to override the enter key other than in the core.

Reply:
Moved the enter code to the core, to the `newline/InsertAccordion.ts` to be exactly. Hoping this is a correct way to go.